pkgs/by-name/fa/fastqc/package.nix

```diff
@@ -0,0 +1,77 @@
+{ lib,
+  stdenv,
+  fetchzip,
+  jre,
+  perl,
+  makeWrapper,
+  imagemagick,
+  makeDesktopItem,
+  copyDesktopItems,
+  desktopToDarwinBundle
+}:
+
+stdenv.mkDerivation rec {
+  pname = "fastqc";
+  version = "0.12.1";
+
+  src = fetchzip {
+    url = "https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v${version}.zip";
+    hash = "sha256-TenRG2x8ivJ2HM2ZpLaJShp0yI0Qc6K5lW5/NJFAa1I";
+  };
+
+  phases = [ "installPhase" "fixupPhase" ];
+
+  nativeBuildInputs = [ makeWrapper imagemagick ]
+                      ++ lib.optional stdenv.isLinux copyDesktopItems # postInstallHook
+                      ++ lib.optional stdenv.isDarwin desktopToDarwinBundle; #fixupOutputHook
+  buildInputs = [ jre perl];
+
+  desktopItem = (makeDesktopItem {
+    name = "FastQC";
+    exec = pname;
+    icon = pname;
+    desktopName = "FastQC";
+    comment = meta.description;
+    categories = [ "Science" ];
+  });
+  desktopItems = [ desktopItem ];
+
+  installPhase = ''
+    mkdir -p $out/{bin,FastQC}
+    cp -r $src/* $out/FastQC
+
+    # Create desktop item
+    mkdir -p $out/share/{applications,icons}
+    # Freedesktop doesn't support windows ICO files. Use imagemagick to convert it to PNG
+    convert $out/FastQC/fastqc_icon.ico $out/share/icons/fastqc.png
+
+    runHook postInstall
+  '';
+
+  fixupPhase = ''
+    patchShebangs $out/FastQC
```

Comment:
I guess `patchShebangs` is executed in default `fixupPhase`.
https://nixos.org/manual/nixpkgs/stable/#ssec-fixup-phase

Reply:
I am not sure, but I can definitely test it and remove that line if necessary.